client/lib/query-manager/activity/index.js

```diff
@@ -0,0 +1,78 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { overEvery, get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import QueryManager from 'lib/query-manager';
+
+/**
+ * ActivityQueryManager manages Activity which can be queried
+ */
+export default class ActivityQueryManager extends QueryManager {
+	constructor( data, options ) {
+		super( data, {
+			itemKey: 'ts_utc',
+			...options,
+		} );
+	}
+
+	/**
+	 * Sort descending order, defaulting to end of list if unknown.
+	 *
+	 * @param  {Object} query Query object (Unused).
+	 * @param  {Object} itemA First item
+	 * @param  {Object} itemB Second item
+	 * @return {Number}       0 if equal, less than 0 if itemA is first,
+	 *                        greater than 0 if itemB is first.
+	 */
+	compare( query, { ts_utc: tsA }, { ts_utc: tsB } ) {
+		return tsB - tsA;
+	}
+
+	/**
+	 * Returns true if the item matches the given query, or false otherwise.
+	 *
+	 * @param  {Object}  query Query object
+	 * @param  {Object}  item  Item to consider
+	 * @return {Boolean}       Whether item matches query
+	 */
+	matches = ActivityQueryManager.matches;
```

Comment:
I find this a bit confusing. Can we absorb it into [L52](https://github.com/Automattic/wp-calypso/pull/17423/files#diff-2ed53d619959b7d5893910f93b5dba0bR52)?